databricks/koalas/internal.py

```diff
@@ -657,96 +672,89 @@ def offset(id):
             (sdf[offset_column] + sdf[row_number_column] - 1).alias(column_name), *scols
         )
 
-    @lazy_property
-    def _column_labels_to_name(self) -> Dict[Tuple[str, ...], str]:
-        return dict(zip(self.column_labels, self.data_columns))
-
-    def column_name_for(self, column_labels_or_index_column: Union[str, Tuple[str, ...]]) -> str:
+    def spark_column_name_for(self, labels: Union[str, Tuple[str, ...]]) -> str:
         """ Return the actual Spark column name for the given column name or index. """
-        if column_labels_or_index_column in self._column_labels_to_name:
-            return self._column_labels_to_name[column_labels_or_index_column]
+        column_labels_to_name = dict(zip(self.column_labels, self.data_spark_column_names))
+        if labels in column_labels_to_name:
+            return column_labels_to_name[labels]  # type: ignore
         else:
-            if column_labels_or_index_column not in self.index_columns:
-                raise KeyError(name_like_string(column_labels_or_index_column))
-            return column_labels_or_index_column  # type: ignore
-
-    @lazy_property
-    def _column_labels_to_scol(self) -> Dict[Tuple[str, ...], spark.Column]:
-        return dict(zip(self.column_labels, self.column_scols))
+            for index_spark_column_name, index_name in self.index_map.items():
+                if index_name == labels:
+                    return index_spark_column_name
```

Comment:
Actually I disagree that this method searches by labels (and index name) because multiple index names could be `None`, then it will return a wrong column name.

Reply:
True.. I will better have to remove this branch.

Reply:
Let me reply on here https://github.com/databricks/koalas/pull/1331/files/13fd0d289a7396c3adc0b78433893040be7202c8#r389330455. Seems it's the similar topic.

Reply:
Maybe I should throw an exception if multiple columns are selected ...